server/game/gamesteps/conflict/conflictflow.js

```diff
@@ -171,6 +172,23 @@ class ConflictFlow extends BaseStepWithPipeline {
         this.game.openEventWindow(events);
     }
 
+    raisePreRevealEvents() {
+        if(this.conflict.conflictPassed) {
+            return;
+        }
+
+        let ring = this.conflict.ring;
+        let events = [this.game.getEvent(EventNames.OnConflictDeclaredBeforeReveal, {
+            conflict: this.conflict,
+            type: this.conflict.conflictType,
+            ring: ring,
+            attackers: this.conflict.attackers.slice(),
+            ringFate: ring.fate
+        })];
+
+        this.game.openEventWindow(events);
```

Comment:
This isn't really the right way to do this - we need to make conflict declaration a proper event which can be interrupted and reacted to.  We need to know where covert fits in to do it correctly though